server/core/src/https/views/admin/accounts.rs

```diff
@@ -0,0 +1,216 @@
+) -> axum::response::Result<Response> {
+    let accounts = get_accounts_info(state, &kopid, client_auth_info, domain_info.clone()).await?;
+    let accounts_partial = AccountsPartialView { accounts };
+
+    let push_url = HxPushUrl(Uri::from_static("/ui/admin/accounts"));
+    Ok(if is_htmx {
+        (push_url, accounts_partial).into_response()
+    } else {
+        (
+            push_url,
+            AccountsView {
+                access_info: AccessInfo::new(),
+                navbar_ctx: NavbarCtx { domain_info },
+                partial: accounts_partial,
+            },
+        )
+            .into_response()
+    })
+}
+
+async fn get_account_info(
+    uuid: Uuid,
+    state: ServerState,
+    kopid: &KOpId,
+    client_auth_info: ClientAuthInfo,
+    domain_info: DomainInfoRead,
+) -> Result<AccountInfo, ErrorResponse> {
+    let scim_entry: ScimEntryKanidm = state
+        .qe_r_ref
+        .scim_entry_id_get(
+            client_auth_info.clone(),
+            kopid.eventid,
+            uuid.to_string(),
+            EntryClass::Account,
+            ScimEntryGetQuery {
+                attributes: Some(vec![
+                    Attribute::Uuid,
+                    Attribute::Description,
+                    Attribute::Name,
+                    Attribute::DisplayName,
+                    Attribute::Spn,
+                    Attribute::Mail,
+                ]),
+                ext_access_check: true,
+            },
+        )
+        .map_err(|op_err| HtmxError::new(kopid, op_err, domain_info.clone()))
+        .await?;
+
+    if let Some(account_info) = scimentry_into_accountinfo(scim_entry) {
+        Ok(account_info)
+    } else {
+        Err(HtmxError::new(kopid, OperationError::InvalidState, domain_info.clone()).into())
+    }
+}
+
+async fn get_accounts_info(
+    state: ServerState,
+    kopid: &KOpId,
+    client_auth_info: ClientAuthInfo,
+    domain_info: DomainInfoRead,
+) -> Result<Vec<AccountInfo>, ErrorResponse> {
+    let filter = filter_all!(f_and!([f_eq(Attribute::Class, EntryClass::Account.into())]));
+    let attrs = Some(BTreeSet::from([
+        Attribute::Uuid,
+        Attribute::Description,
+        Attribute::Name,
+        Attribute::DisplayName,
+        Attribute::Spn,
+        Attribute::Mail,
+    ]));
+    let base: Vec<ScimEntryKanidm> = state
+        .qe_r_ref
+        .scim_entry_search(client_auth_info.clone(), filter, kopid.eventid, attrs, true)
+        .map_err(|op_err| HtmxError::new(kopid, op_err, domain_info.clone()))
+        .await?;
+
+    // TODO: inefficient to sort here
+    let mut accounts: Vec<_> = base
+        .into_iter()
+        // TODO: Filtering away unsuccessful entries may not be desired.
+        .filter_map(scimentry_into_accountinfo)
+        .collect();
+
+    accounts.sort_by_key(|gi| gi.uuid);
+    accounts.reverse();
+    Ok(accounts)
+}
+
+fn scimentry_into_accountinfo(scim_entry: ScimEntryKanidm) -> Option<AccountInfo> {
```

Comment:
This can probablyf be "Impl TryFrom<ScimEntryKanidm> on AccountInfo.". Though isn't there already a ScimPerson or something we crould use?

Reply:
ScimSyncPerson exists but it doesn't contain other fun attributes like what groups they're a member of, and it doesn't use ScimMail for emails strangely.

Also kind of ties into my last question `I think I want to show all attributes in a generic way instead of the hardcoding that I was doing right now, thoughts ?`.
  with other words: after rebasing I started to feel the urge to remove AccountInfo and just pass the attribute map to askama. But that makes a nice layout harder I suppose.

Reply:
Don't re-use scimSyncPerson, that's for sync accounts. 

Make something that is presentable as a person/account. I think that's a better way. That's why there is https://github.com/kanidm/kanidm/blob/master/proto/src/scim_v1/server.rs which is server to client entry types. Make a ScimPerson there.